# Homework 4

In [2]:
# import common
import pandas as pd
import numpy as np
from scipy import stats

import seaborn as sns
import matplotlib.pylab as plt
from statsmodels.stats import weightstats
from statsmodels.stats import proportion

#### 1. 대한민국 성인의 대학교육 이수 비율은 49%이다. 만일 대한민국 성인 중 200명을 무작위 추출한다면 이 표본에서 대졸자의 비율이 45% 미만일 확률은 무엇인가? (표본분포를 이용하여 답을 구하시오.)

In [3]:
n = 200 # 무작위 추출
p = 0.49 # 이수비율
p_se = np.sqrt(p * (1 - p) / n) # 표준편차(표준오차)
stats.norm.cdf(0.45, loc = p, scale = p_se)

0.1289019105950467

---------------

#### 2. `airbnb.csv` 데이터셋을 사용하여 샌프란시스코 전체 리스팅의 `reviews_per_month`의 평균을 아래의 방법으로 추정하시오. 


#### (a) 99% 신뢰구간을 강의노트에 있는 신뢰구간 공식을 사용하여 구하시오. 

In [4]:
df = pd.read_csv("./data/airbnb.csv")
df.reviews_per_month # 데이타 살펴보고

0       1.77
1       1.33
2       0.87
3       0.15
4       0.13
        ... 
6100    1.00
6101    1.00
6102    1.00
6103    1.00
6104    0.23
Name: reviews_per_month, Length: 6105, dtype: float64

In [7]:

n = df.reviews_per_month.count()  # 표본갯수
n

6105

In [8]:
X_bar = df.reviews_per_month.mean()  # 표본평균
X_bar

1.8882506142506144

In [9]:
S = df.reviews_per_month.std() # 표본표준편차
S

2.0899042145301565

In [10]:
from scipy.stats import t
T = t.ppf(0.995, n - 1)
T

2.5766350039580046

In [11]:
X_bar - T * S / np.sqrt(n), X_bar + T * S / np.sqrt(n)

(1.819332013710772, 1.9571692147904567)

-----------

#### (b) 99% 신뢰구간을 `statsmodels.stats.weightstats` 의 `tconfint_mean` 함수를 사용하여 구해서 위에서 직접 계산한 신뢰구간과 일치하는지 확인하시오. 

In [12]:
stats = weightstats.DescrStatsW(df.reviews_per_month)
stats.tconfint_mean(alpha=0.01)  # 이것은 99% 신뢰구간

(1.819332013710772, 1.9571692147904567)

- 일치합니다.  

--------------------

#### (c) 위에서 구한 신뢰구간을 해석하시오. 

- AirBNB이용자의 월 평균 리뷰 수(reviews_per_month)가 1.819 ~ 1.957 사이일 가능성이 99%이다.  

--------------------------

#### 3. 췌장암에 대한 진단은 제1종 오류가 약 25%이다. 이 말의 의미를 설명하라.  이 경우 제1종 오류와 제2종 오류 중 어느 것이 더 위험한 결과를 초래하는가?

제 1종 오류는 귀무가설이 사실임에도 기각되는 오류로  
문제에서 제시된 경우의 귀무가설은 "췌장암에 걸리지 않았다."로 할 수 있으며,  
이와 관련하여 제 1종 오류가 약 25%는 "췌장암에 걸리지 않았는데, 걸렸다고 판단하는 오류가 약 25%"라는 의미로 해석할 수 있습니다.  
  
제 2종 오류는 귀무가설이 거짓임에도 채택되는 오류로  
"췌장암에 걸렸는데 췌장암으로 판단하지 않는 경우"이기 때문에 생명과 관련한 본 문제에서는 제 2종 오류가 더 위험한 결과를 초래한다고 볼 수 있습니다.  

----------------------